keys/keys.go

```diff
@@ -0,0 +1,13 @@
+package keys
+
+// MasterKey provides a way of securing the key used to encrypt the Tree by encrypting and decrypting said key.
+type MasterKey interface {
```

Comment:
I would like to see this interface moved to be more functional. Encrypt in particular - it could take []byte (unencrypted data key) and return []byte (encrypted data key) rather than having implementing structs (kms.MasterKey and pgp.MasterKey) have fields for the encrypted data key.

Reply:
The encrypted data key eventually has to be passed to the stores, because it has to be on the output file. Currently this is simple because MasterKeys store the encrypted data key themselves. I'm not sure how this could be done cleanly.

Reply:
That's a lot of interfaces to satisfy. Are they all required? Can some of them be moved to optional interfaces to make it easier to implement new master keys without requiring all the functionalities?